### DISCUSSION TITLE
Thoughts before sprint

### DISCUSSION THREAD

**Discussion by ndaelman-hu** [FAIRmat-NFDI/nomad-simulations#23] [Ideas]
These are just a bunch of notes from during the DPG 2024, to be applied during the sprint.

**Comment by ndaelman-hu** [comment]
# Dimensionality vs periodicity

There are 3 ways to regulate dimensionality, depending on the restrictions of the code.
1. directly, i.e. `dimensionality`.
2. via periodic boundary conditions (PBC)
3. adding a vacuum along the axis to be excluded.

Currently, option 2 is broken (see below) and there's no option 3.
Lastly, we need an "interpreted / effective dimensionality" that encompasses all 3 options together, making them interoperable (maybe `type`?)

For option 2, we currently have

``` python
    periodic_boundary_conditions = Quantity(
        type=bool,
        shape=[3],
        description="""
        If periodic boundary conditions are applied to each direction of the crystal axes.
        """,
    )
```

My recommendations:
- Abbreviate the name. I think `pbc` is well-known enough.
- The `shape` should not be a constant. Else, we would end up in scenarios where we have 3 PBC axes for true 2D systems... They wouldn't be interpretable anymore.

**Reply by JosePizarro3** [reply]
Ok, feel free to change the naming. About the `shape`, I am not sure if I understand. For 2D system, `pbc = [True, True, False]` anyways. Or am I understanding that the shape should be set by `dimensionality` itself? In that case, they would end up being equivalent quantities.
